File: content/problems/0830-positions-of-large-groups/solutions/scan-maximal-runs/Solution.java
class Solution {
    public List<List<Integer>> largeGroupPositions(String s){
        List<List<Integer>> answer=new ArrayList<>();
        for(int start=0;start<s.length();){
            int end=start+1;
            while(end<s.length()&&s.charAt(end)==s.charAt(start))end++;
            if(end-start>=3)answer.add(Arrays.asList(start,end-1));
            start=end;
        }
        return answer;
    }
}

File: content/problems/0830-positions-of-large-groups/solutions/scan-maximal-runs/solution.cpp
class Solution {
    public: vector<vector<int>> largeGroupPositions(string s){
        vector<vector<int>> answer;
        for(int start=0;start<(int)s.size();){
            int end=start+1;
            while(end<(int)s.size()&&s[end]==s[start])++end;
            if(end-start>=3)answer.push_back({
                start,end-1
            });
            start=end;
        }
        return answer;
    }
};
